pkg/controllers/clusterresourceplacement/controller.go

```diff
@@ -660,3 +676,122 @@ func parseResourceGroupHashFromAnnotation(s *fleetv1beta1.ClusterResourceSnapsho
 	}
 	return v, nil
 }
+
+// TODO: handle multiple resourceSnapshots
+func (r *Reconciler) buildPlacementStatus(crp *fleetv1beta1.ClusterResourcePlacement, latestSchedulingPolicySnapshot *fleetv1beta1.ClusterSchedulingPolicySnapshot, latestResourceSnapshot *fleetv1beta1.ClusterResourceSnapshot) (*fleetv1beta1.ClusterResourcePlacementStatus, error) {
+	status := &fleetv1beta1.ClusterResourcePlacementStatus{
+		SelectedResources: make([]fleetv1beta1.ResourceIdentifier, 0, len(latestResourceSnapshot.Spec.SelectedResources)),
+		PlacementStatuses: make([]fleetv1beta1.ResourcePlacementStatus, 0, 20), // pre-allocate with a reasonable capacity
+		Conditions:        []metav1.Condition{},
+	}
+	// TODO query the sub resourceSnapshots by using latestResourceSnapshot and then update the status.
```

Comment:
work generator already does this, maybe we can refactor it to share the common logic?

Reply:
sure, will use it when handling multiple resourceSnapshots.